.github/workflows/push.yml

```
@@ -38,3 +47,67 @@ jobs:
         env:
           GO111MODULE: "on"
         run: make all
+
+  binaries:
+      name: Build binaries and notarize sources
+      runs-on: ubuntu-latest
+      env:
+        JOB_NAME: ${{ github.job }}
+        JOB_ID: ${{ github.run_id }}
+      outputs:
+        matrix: ${{ steps.list-binaries.outputs.matrix }}
+      steps:
+        - uses: actions/setup-go@v3
+          with:
+            go-version: ${{ env.GO_VERSION }}
+        - uses: actions/checkout@v3
+        - name: Build binaries
+          run: make dist
+        - id: list-binaries
+          run: |
+            echo "::set-output name=matrix::$(ls dist | jq -R -s -c 'split("\n")[:-1] | {binary: .}')"
+        # - name: Upload binary artifacts
+        #   uses: actions/upload-artifact@v3
+        #   with:
+        #     name: immugw-binaries
+        #     path: dist
+        #     retention-days: 5
+        - name: Calculate checksums
+          run: make dist/binary.md
+
+  images:
+      name: Build and notarize Docker Images
+      needs:
+        - binaries
+        - binaries-quick-test
+        - stress-tests
+      runs-on: ubuntu-latest
+      env:
+        JOB_NAME: ${{ github.job }}
+        JOB_ID: ${{ github.run_id }}
+        DOCKER_IMAGE_IMMUGW: "codenotary/immugw"
+        DOCKER_BUILDKIT: "1"
+      steps:
+        - uses: actions/checkout@v3
+        - name: Build docker images
+          shell: bash
+          run: |
+            if [[ "${GITHUB_REF}" =~ refs/tags/v([0-9]+)\.([0-9]+)\.([0-9]+) ]]; then
```

This requires the same fix as in the immudb project:

```suggestion
            if [[ "${GITHUB_REF}" =~ ^refs/tags/v([0-9]+)\.([0-9]+)\.([0-9]+)$ ]]; then
```

---

ack